### PR TITLE
New user helper extensions.

### PR DESCRIPTION
**Pull request by igomez06**
Fixes from last PR are here as well

### CODE REVIEW

File: tests/framework/clients/rancher/client.go [hunk 1]
```diff
@@ -71,6 +71,21 @@ func clientOpts(restConfig *rest.Config, rancherConfig *Config) *clientbase.Clie
...
+	token := &management.Token{
+		UserID: user.ID,
+	}
+
+	returnedToken, err := c.Management.Token.Create(token)
+	if err != nil {
+		return nil, err
+	}
+
+	bearerToken := returnedToken.Token
```
Comment (paynejacob) [inline tests/framework/clients/rancher/client.go]: there is no need to copy the token here

File: tests/framework/extensions/users/users.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import (
+	"time"
+
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+func CreateUserWithRole(rancherClient *rancher.Client, user *management.User, role string) (*management.User, error) {
```
Comment (paynejacob) [inline tests/framework/extensions/users/users.go]: just a thought would this be more useful if role was a variadic parameter?

File: tests/framework/extensions/users/users.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+}
+
+func AddProjectMember(rancherClient *rancher.Client, project *management.Project, user *management.User, projectRole string) error {
+	role := &management.ProjectRoleTemplateBinding{
+		ProjectID:       project.ID,
+		UserPrincipalID: user.PrincipalIDs[0],
+		RoleTemplateID:  projectRole,
+	}
+
+	err := wait.Poll(1*time.Second, 1*time.Minute, func() (bool, error) {
```
Comment (paynejacob) [inline tests/framework/extensions/users/users.go]: we can assume the project exists here we need to wait for the condition `BackingNamespaceCreated`. We should also use a watch.

**Review by paynejacob — CHANGES_REQUESTED** [review]

File: tests/framework/extensions/users/users.go (hunk 2 above)
Comment (igomez06) [reply tests/framework/extensions/users/users.go]: Hmmm, how would that work in this function. We create multiple GlobalRoleBindings for a single user?

Comment (paynejacob) [reply tests/framework/extensions/users/users.go]: yeah we would just loop through the roles and add each one

Comment (igomez06) [reply tests/framework/extensions/users/users.go]: If you think it's necessary. I can make the changes for sure

File: tests/framework/clients/rancher/client.go (hunk 1 above)
Comment (igomez06) [reply tests/framework/clients/rancher/client.go]: fixed

File: tests/framework/extensions/users/users.go (hunk 2 above)
Comment (igomez06) [reply tests/framework/extensions/users/users.go]: updated

File: tests/framework/extensions/users/users.go (hunk 3 above)
Comment (igomez06) [reply tests/framework/extensions/users/users.go]: fixed

File: tests/framework/extensions/users/users.go [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+		}
+	}
+
+	return createdUser, nil
+}
+
+func AddProjectMember(rancherClient *rancher.Client, project *management.Project, user *management.User, projectRole string) error {
+	role := &management.ProjectRoleTemplateBinding{
+		ProjectID:       project.ID,
+		UserPrincipalID: user.PrincipalIDs[0],
```
Comment (paynejacob) [inline tests/framework/extensions/users/users.go:42]: are users guaranteed to have at least 1 principal?

File: tests/framework/extensions/users/users.go [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+		RoleTemplateID:  projectRole,
+	}
+
+	name := strings.Split(project.ID, ":")
```
Comment (paynejacob) [inline tests/framework/extensions/users/users.go]:
```suggestion
	name := strings.Split(project.ID, ":")[1]
```

File: tests/framework/extensions/users/users.go [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+	opts := metav1.ListOptions{
+		FieldSelector:  "metadata.name=" + name[1],
```
Comment (paynejacob) [inline tests/framework/extensions/users/users.go]:
```suggestion
		FieldSelector:  "metadata.name=" + name,
```

**Review by paynejacob — APPROVED** [review]
minor nits looks great.

File: tests/framework/extensions/users/users.go (hunk 4 above)
Comment (igomez06) [reply tests/framework/extensions/users/users.go:42]: Yes they are

**Pull request merged**